File: backend/DTOs/Requests/Auth/RegisterRequest.cs
namespace DataLabellingSupportSystem.Api.DTOs.Requests.Auth;

public sealed record RegisterRequest(
    string FullName,
    string Email,
    string Password,
    string? PhoneNumber,
    string? IdentifyNumber,
    string? Gender,
    string? Address,
    DateOnly? DateOfBirth
);

File: mobile_app/lib/models/auth/register_request.dart
class RegisterRequest {
  final String fullName;
  final String email;
  final String password;
  final String? phoneNumber;
  final String? identifyNumber;
  final String? gender;
  final String? address;
  final DateTime? dateOfBirth;

  RegisterRequest({
    required this.fullName,
    required this.email,
    required this.password,
    this.phoneNumber,
    this.identifyNumber,
    this.gender,
    this.address,
    this.dateOfBirth,
  });

  Map<String, dynamic> toJson() => {
        'fullName': fullName,
        'email': email,
        'password': password,
        'phoneNumber': phoneNumber,
        'identifyNumber': identifyNumber,
        'gender': gender,
        'address': address,
        'dateOfBirth': dateOfBirth?.toIso8601String(),
      };
}
